src/Adapter/Product/Stock/CommandHandler/UpdateProductStockInformationHandler.php

```diff
@@ -43,25 +45,40 @@ final class UpdateProductStockInformationHandler implements UpdateProductStockIn
      */
     private $productStockUpdater;
 
+    /**
+     * @var MovementReasonRepository
+     */
+    private $movementReasonRepository;
+
     /**
      * @param ProductStockUpdater $productStockUpdater
+     * @param MovementReasonRepository $movementReasonRepository
      */
     public function __construct(
-        ProductStockUpdater $productStockUpdater
+        ProductStockUpdater $productStockUpdater,
+        MovementReasonRepository $movementReasonRepository
     ) {
         $this->productStockUpdater = $productStockUpdater;
+        $this->movementReasonRepository = $movementReasonRepository;
     }
 
     /**
      * {@inheritdoc}
      */
     public function handle(UpdateProductStockInformationCommand $command): void
     {
+        $deltaQty = null;
+        if (null !== $command->getDeltaQuantity()) {
```

Comment:
```suggestion
        if (null === $command->getDeltaQuantity()) {
               return;
        }
```

Reply:
And then do your treatment

Reply:
same as above. I cannot return here as there are other props that might need an update.

Reply:
Yes, you're right! Thanks